### ISSUE TITLE
Dropped items are disappearing

### ISSUE THREAD

**Issue by pjindras** [Dreeam-qwq/Leaf#173]
### Checklist

- [X] I have searched the existing issues
- [ ] I ticked all the boxes without actually reading them
- [X] I am using latest Leaf build from https://github.com/Winds-Studio/Leaf/releases
- [X] I tested and couldn't replicate this issue on [latest Purpur build](https://purpurmc.org/)

### What behaviour is observed?

When I take item from itemframe, sometimes it disappears immediately. Sometimes the item is dropped correctly.

https://github.com/user-attachments/assets/44513435-20fc-41d9-a0e0-018b6b38157b

### Expected behaviour

Item should stay there 6000 ticks as I have defined in spigot.yml.

### To Reproduce

Itemframe (maybe there must be more entities around) put in put out -> after some time, it happens.
Mostly it happens if the itemframe is between armorstands.

There is no plugin, which could cause it. The item has nbt tags, enchants etc, but non of theme is incorrect. If I drop the item on the ground, its usually ok (it doesn't disappear)

Edit: it can be a vanilla item (without tags, etc), but it disappears when the itemframe is between more armor stands.

So put armorstand itemframe armorstand -> take the item and put it in -> out -> in and 5-10 times the same and the item will disappear.

### Spark profiler link

https://spark.lucko.me/JWyP4EVXf2

**Comment by Dreeam-qwq** [comment]
Can you try to replicate on a clean Leaf server without plugins?

**Comment by pjindras** [comment]
> Can you try to replicate on a clean Leaf server without plugins?

Yes, but it is more complicated than I thought,

There must be near this cart, which is flying on slime block :D like this.

If this is near or something similar, than the items from itemframe disappears.


https://github.com/user-attachments/assets/83892f9e-4c24-4bc7-a1ac-cc57c32efceb

I know that this is strange but we spent a lot of time on testing before we figure that out.

**Comment by Dreeam-qwq** [comment]
Can you try to disable optimize-minecart in leaf config?

**Comment by pjindras** [comment]
Yes, but unfortunately it had no effect. It still happens with optimize-minecart disabled but I will try to investigate more.

**Comment by Dreeam-qwq** [comment]
I guess this issue is not on Leaf 1.21.3+ right?

**Comment by Taiyou06** [comment]
@pjindras is this present in newer versions? if not gonna close this issue.

**Comment by pjindras** [comment]
ok, its fixed now
